modin/data_management/utils.py

```diff
@@ -82,15 +82,16 @@ def split_result_of_axis_func_pandas(axis, num_splits, result, length_list=None)
     list
         A list of Pandas DataFrames.
     """
-    if num_splits == 1:
-        return result
     if length_list is not None:
         length_list.insert(0, 0)
         sums = np.cumsum(length_list)
         if axis == 0:
             return [result.iloc[sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
         else:
             return [result.iloc[:, sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
+
+    if num_splits == 1:
+        return [result]
```

Comment:
Why do we wrap to a list?

Reply:
This is a simplification - now `split_result_of_axis_func_pandas` function returns `list` everywhere (as it is written in the docstring), which in turn localizes this edge case in this function, and not in several.